aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -441,6 +444,150 @@ std::tuple<Tensor&,Tensor&> min_out(Tensor& min, Tensor& min_indices,
   return result;
 }
 
+static Tensor scalar_tensor_like(const Tensor& like, const Scalar& value) {
+  // FIXME: This should do type promotion with value, but that would be a BC-break
+  return at::scalar_tensor(value, at::device(like.device()).dtype(like.scalar_type()));
```

Comment:
We have so many ways to create tensors from scalars (and I should really take the time to fix them)... Won't this cause a cross-device sync, though?

Reply:
Agree, this is a serious perf issue. 
We don't actually have many ways of creating device tensors, it's `scalar_tensor` and `wrapped_scalar_tensor` that have different behavior wrt type promotion.